platform/osgi-web-util/src/main/java/org/motechproject/osgi/web/ModuleRegistrationData.java

```diff
@@ -207,6 +213,20 @@ public void setSubMenu(Map<String, SubmenuInfo> subMenu) {
     }
 
     /**
+     * Returns the map of available tabs for specified permissions for this module.
+     * @return a map, where the keys are permissions names and values are lists of accessible tabs for that permission
+     */
+    @JsonIgnore
+    public Map<String, List<String>> getTabAccessMap() { return tabAccessMap; }
+
+    /**
+     * Sets the map available tabs for specified permissions for this module.
+     * @param tabAccessMap a map, where the keys are permissions names and values are lists of accessible tabs for that permission
+     */
+    @JsonIgnore
+    public void setTabAccessMap(Map<String, List<String>> tabAccessMap) { this.tabAccessMap = tabAccessMap; }
```

Comment:
I would rather expect tabs to be the keys, and permission list to be the values. Why do it in this backward fashion? Moreover, are we supporting a case where we have a tab without any permissions required? Flipping keys and values will probably be required for that.